src/Negotiation/AbstractNegotiator.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Negotiation;
+
+abstract class AbstractNegotiator
+{
+    /**
+     * @param string $header     A string containing an `Accept|Accept-*` header.
+     * @param array  $priorities A set of server priorities.
+     *
+     * @return BaseAccept best matching type
+     */
+    public function getBest($header, array $priorities)
+    {
+        if (!$priorities) {
+            throw new \Exception('no priorities given'); 
```

Comment:
```
$ grep -R Exception .
./AcceptLanguage.php:            throw new \Exception('invalid language type in header.');
./AbstractNegotiator.php:            throw new \Exception('no priorities given'); 
./AbstractNegotiator.php:            throw new \Exception('empty header given'); 
./AbstractNegotiator.php:            throw new \Exception('failed to parse accept header');
./Accept.php:            throw new \Exception('invalid media type.');
```
Seems there are 2 classes of exception:
  * errors from parsing the header
  * errors from bad arguments passed from the application